### PR TITLE
Closes #6372: Log on submit status last entry

### LINKED ISSUE #6372: As a support team member,  I want to know the last time an entry was added to wpr_rucss_used_css table with the status on-submit so that I can check if the RUCSS feature is properly triggered
**Context**

As part of building a RUCSS Debug tool, some information must be stored while in WP Rocket Debug mode so that it can be monitored easily by the support team.
More details here: https://www.notion.so/wpmedia/RUCSS-Issues-7adc4ed750fd4dc7902bb6881ed1204b?pvs=4#38398060400d4fcf97f27aa563483d68

**Expected behavior**
- When in debug mode (WP_ROCKET_DEBUG?), a timestamp must be updated in the DB every time the `UsedCSS::create_new_job` method is successful. We only need to keep the last time it happened.
- When not in debug mode, nothing special happens. If there is a timestamp stored, it can stay there.

### PR DESCRIPTION
**Pull request by jeawhanlee**
## Description

This PR logs the last entry for new rucss job added.

Fixes #6372 

## Type of change

- New feature (non-breaking change which adds functionality).

## Is the solution different from the one proposed during the grooming?

No

# Checklists

## Generic development checklist

- [ ] My code follows the style guidelines of this project, with adapted comments and without new warnings.
- [ ] I have added unit and integration tests that prove my fix is effective or that my feature works.
- [ ] The CI passes locally with my changes (including unit tests, integration tests, linter).
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] If applicable, I have made corresponding changes to the documentation. *Provide a link to the documentation.*

## Test summary

- [ ] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [ ] I validated all Acceptance Criteria of the related issues. (If applicable, provide proof).
- [ ] I validated all test plan the QA Review asked me to.

*If not, detail what you could not test.*

*Please describe any additional tests you performed.*

### CODE REVIEW

File: inc/Engine/Debug/RUCSS/Subscriber.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace WP_Rocket\Engine\Debug\RUCSS;
+
+use WP_Rocket\Admin\Options_Data;
```
Comment (remyperona) [inline inc/Engine/Debug/RUCSS/Subscriber.php]:
```suggestion
use WP_Rocket\Admin\{Options, Options_Data};
```

File: inc/Engine/Debug/Resolver.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	public static function get_services(): array {
+		$set_services = [];
+
+		if ( empty( self::$options ) ) {
+			return [];
+		}
+
+		foreach ( self::$options as $option => $services ) {
+
+			if ( ! (bool) get_rocket_option( $option ) ) {
```
Comment (remyperona) [inline inc/Engine/Debug/Resolver.php]: We could use the Options_Data instance instead

File: inc/classes/ServiceProvider/class-options.php [hunk 3]
```diff
@@ -22,6 +22,7 @@ class Options extends AbstractServiceProvider {
 	 */
 	protected $provides = [
 		'options',
+		'options_debug',
```
Comment (remyperona) [inline inc/classes/ServiceProvider/class-options.php]: I would register this class in the Debug service provider too instead

**Review by remyperona — CHANGES_REQUESTED** [review]

File: wp-rocket.php [hunk 4]
```diff
@@ -26,6 +26,7 @@
 define( 'WP_ROCKET_PHP_VERSION',           '7.3' );
 define( 'WP_ROCKET_PRIVATE_KEY',           false );
 define( 'WP_ROCKET_SLUG',                  'wp_rocket_settings' );
+define( 'WP_ROCKET_DEBUG_SLUG',            'wp_rocket_debug' );
```
Comment (remyperona) [inline wp-rocket.php]: I think we should move away from using global constants like this. We can use the option name directly, with our architecture now it's more centralized.

File: inc/admin/upgrader.php [hunk 5]
```diff
@@ -143,6 +143,21 @@ function rocket_first_install() {
...
+		 *
+		 * @param array Array of default rocket debug options.
+		 */
+		apply_filters(
+			'rocket_first_install_debug_options',
+			[
+				'last_rucss_job_added' => '',
+			]
+		)
+	);
```
Comment (remyperona) [inline inc/admin/upgrader.php]: Couple of things here:
- We should do this in a separate method in a debug subscriber
- We need to check that it's only done once when upgrading, by using the plugin version check

File: inc/Engine/Debug/ServiceProvider.php [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+			return;
+		}
+
+		$this->container->add( 'options_debug', Options_Data::class )
+			->addArgument( $this->container->get( 'options_api' )->get( 'debug', [] ) );
+
+		foreach ( $services as $service ) {
+			$this->getContainer()->add( $service['service'], $service['class'] )
+				->addArgument( $this->getContainer()->get( 'options_debug' ) )
+				->addArgument( $this->getContainer()->get( 'options_api' ) );
```
Comment (remyperona) [inline inc/Engine/Debug/ServiceProvider.php:69]: this only works if all services have the same construct signature forever, are we sure about it?

File: inc/Engine/Debug/Resolver.php [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+class Resolver {
+
+	/**
+	 * Array of WP Rocket Options.
+	 *
+	 * @var array
+	 */
+	private static $options_services = [
+		'remove_unused_css' => [
+			'service' => 'rucss_subscriber',
```
Comment (remyperona) [inline inc/Engine/Debug/Resolver.php]:
```suggestion
			'service' => 'rucss_debug_subscriber',
```

**Review by remyperona — CHANGES_REQUESTED** [review]

File: inc/Engine/Debug/ServiceProvider.php (hunk 6 above)
Comment (jeawhanlee) [reply inc/Engine/Debug/ServiceProvider.php:69]: From what I see, we'll mostly be updating the debug options based on triggered hooks. Do you see something else?

File: inc/Engine/Debug/Resolver.php [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+		foreach ( self::$options_services as $option => $services ) {
+			if ( ! (bool) self::$options->get( $option, 0 ) ) {
+				continue;
+			}
+
+			$set_services[] = $services;
+		}
+
+		return $set_services;
+	}
```
Comment (CrochetFeve0251) [inline inc/Engine/Debug/Resolver.php]: @jeawhanlee why everything is static?

File: inc/Engine/Debug/RUCSS/Subscriber.php [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+	 * @param string $timestamp Current timestamp.
+	 * @return void
+	 */
+	public function log_last_added_job_time( $is_success, $timestamp ) {
+		if ( ! $is_success ) {
+			return;
+		}
+
+		$this->options->set( 'last_rucss_job_added', $timestamp );
+		$this->options_api->set( 'debug', $this->options->get_options() );
```
Comment (CrochetFeve0251) [inline inc/Engine/Debug/RUCSS/Subscriber.php:59]: @jeawhanlee by default options contains the wpr settings.

**Review by remyperona — APPROVED** [review]

**Pull request merged**